src/api/api.ts

```diff
@@ -0,0 +1,56 @@
+
+export interface Good{
+  categoryTypeId: string,
+  id:string,
+  img:string,
+  label:string,
+  price:number,
+}
+export interface Category {
+   type:string,
+   label:string,
+   id:string,
+}
+
+export class Api {
+
+  endPoints={
+    goods:'/api/goods',
+    categories:'/api/categories',
+    popular_categories:'/api/popular_categories',
+  
+  }
+  
+    request = (url:string, params?:{categoryTypeIds:string,ids?:string, text?:string}) => {
```

Comment:
```suggestion
    request = (url:string, params?: Record<string, any>) => {
```
request общая функция которая делает запросы с любыми квери параметрами, для нее главное - чтобы params был объект, а с какими значениями - его не волнует